src/cpp/handler/variantutil.cpp

```diff
@@ -271,8 +274,8 @@ bool convertToJsonStyleInPlace(QVariant *in)
 
 	bool changed = false;
 
-	int type = in->type();
-	if(type == QVariant::Hash)
+	int type = typeId(*in);
```

Comment:
Should this be `QMetaType::Type`, not `int`?  I think you've made that substitution elsewhere.